ipatests/test_integration/test_trust.py

```diff
@@ -463,6 +357,50 @@ def test_user_uid_resolution_in_external_trust_with_shared_secret(self):
         assert re.search(
             testuser_regex, result.stdout_text), result.stdout_text
 
+    def test_override_homedir(self):
+        """
+        https://pagure.io/SSSD/sssd/issue/2474
+        POSIX attributes, such as shell or home directory, should not be overwritten or missing.
+        """
+        tasks.establish_trust_with_ad(self.master, self.ad_domain,
+                                      extra_args=['--range-type', 'ipa-ad-trust'])
+
+        tasks.backup_file(self.master, paths.SSSD_CONF)
+        testuser = 'testuser@%s' % self.ad_domain
+        result1 = self.master.run_command(['getent', 'passwd', testuser])
+        assert "/home/%s/testuser" % self.ad_domain in result1.stdout_text
+
+        domain = self.master.domain
+        tasks.modify_sssd_conf(
+            self.master,
+            domain.name,
+            {
+                'subdomain_homedir': '%o'
+            }
+        )
+        sssd_conf = self.master.get_file_contents(
+            paths.SSSD_CONF, encoding='utf-8'
+        )
+        assert "subdomain_homedir" in sssd_conf
+        tasks.clear_sssd_cache(self.master)
+
+        run_cmd = '"import os; os.initgroups({0}, 1234)"' .format(testuser)
+        self.master.run_command(['python', '-c', run_cmd])
+
+        testuser = 'testuser@%s' % self.ad_domain
+        result2 = self.master.run_command(['getent', 'passwd', testuser])
+
+        assert "/home/testuser" in result2.stdout_text
+
+        log_file = "/var/log/sssd/sssd_{0}.log" .format(self.master.domain.name)
+        sssd_debug_log = self.master.get_file_contents(
+            log_file, encoding='utf-8'
+        )
+        assert 'get_subdomain_homedir_of_user failed' not in sssd_debug_log
+
+        tasks.restore_files(self.master)
```

Comment:
The cleanup will not be executed if test fails at some assertion. To make it run regardless of test success you can put the test in "try" block and cleanup statements in "finally" block

Reply:
If you setup trust in the testcase, you should also remove it in teardown